ananse/network.py

```diff
@@ -73,89 +75,36 @@ def __init__(self, ncore=1, genome="hg38", gene_bed=None, include_promoter=False
+
         # self.promoter = promoter
         self.include_promoter = include_promoter
 
         self.include_enhancer = include_enhancer
 
-    def clear_peak(self, ddf):
-        """
-        Filter the enhancer peaks in promoter range.
-        """
-        ddf = ddf.compute()
-
-        global alltfs
-        alltfs = list(set(ddf.factor))
-
-        enhancerbed = pd.DataFrame(set(ddf.enhancer))
-        enhancerbed[["chr","site"]]=enhancerbed[0].str.split(":",expand=True)
-        enhancerbed[["start","end"]]=enhancerbed.site.str.split("-",expand=True)
-        enhancerbed.drop(columns=[0,"site"], inplace=True)
-
-        enhancerfile = NamedTemporaryFile(mode="w", dir=mytmpdir(), delete=False)
-        enhancerbed.to_csv(enhancerfile, sep="\t", header=False, index=False)
-        # print(enhancerfile.name)
-        return alltfs, enhancerfile.name
-
-    def clear_peak_df(self, ddf):
-        """
-        Filter the enhancer peaks in promoter range.
-        """
-
-        global alltfs
-        alltfs = list(set(ddf.factor))
-
-        enhancerbed = pd.DataFrame(set(ddf.enhancer))
-        enhancerbed[["chr","site"]]=enhancerbed[0].str.split(":",expand=True)
-        enhancerbed[["start","end"]]=enhancerbed.site.str.split("-",expand=True)
-        enhancerbed.drop(columns=[0,"site"], inplace=True)
-
-        enhancerfile = NamedTemporaryFile(mode="w", dir=mytmpdir(), delete=False)
-        enhancerbed.to_csv(enhancerfile, sep="\t", header=False, index=False)
-        # print(enhancerfile.name)
-        return alltfs, enhancerfile.name
-
-    def get_promoter_dataframe(self, peak_bed, up=2000, down=2000):
-        # all overlap Enh-TSS(up2000 to down2000) pair
-        peaks = BedTool(peak_bed)
-        b = BedTool(self.gene_bed)
-        b = b.flank(l=1, r=0, s=True, g=self.gsize).slop(  # noqa: E741
-            l=up, r=down, g=self.gsize, s=True  # noqa: E741
-        )
-        vals = []
-        # for f in b.intersect(peaks, wo=True, nonamecheck=True):
-        for f in b.intersect(peaks, wo=True):
-            chrom = f[0]
-            gene = f[3]
-            peak_start, peak_end = int(f[13]), int(f[14])
-            vals.append([chrom, gene, peak_start, peak_end])
-        prom = pd.DataFrame(vals, columns=["chrom", "gene", "peak_start", "peak_end"])
-        prom["loc"] = (
-            prom["chrom"]
-            + ":"
-            + prom["peak_start"].astype(str)
-            + "-"
-            + prom["peak_end"].astype(str)
-        )
-        prom.gene = [i.upper() for i in list(prom.gene)]
-        return prom
-
-    def get_gene_dataframe(self, peak_bed, up=100000, down=100000):
+    def get_gene_dataframe(
+        self,
+        peak_bed,
+        up=100000,
+        down=100000,
+        alpha=1e4,
+        promoter=2000,
+        full_weight_region=5000,
+    ):
         # all overlap Enh-TSS(100000-tss-100000) pair with distance
+        logger.info("relating enhancers to genes")
         peaks = BedTool(peak_bed)
```

Comment:
p.s. another check if there are only unique genes in here? In e.g. xt10 there are multiple of the same gene

https://raw.githubusercontent.com/vanheeringen-lab/ANANSE/master/ananse/db/xt10.0.gene.bed (e.g. efna3)

Not sure what the effects of that is downstream